### ISSUE TITLE
💅 organizeImports reported on first import for unsorted exports

### ISSUE THREAD

**Issue by Kakadus** [biomejs/biome#9530]
### Environment information

Reproduced on playground.

### Rule name

organizeImports

### Playground link

https://biomejs.dev/playground/?lintRules=organizeImports&cssModules=true&code=aQBtAHAAbwByAHQAIAAiAGEAIgA7AAoAaQBtAHAAbwByAHQAIAAiAGIAIgA7AAoACgAKAGMAbwBuAHMAdAAgAGEAIAA9ACAAIgBhACIAOwAKAGMAbwBuAHMAdAAgAGIAIAA9ACAAIgBiACIAOwAKAAoAZQB4AHAAbwByAHQAIAB7ACAAYgAsACAAYQAgAH0AOwAKAA%3D%3D&language=ts

### Expected result

The error / warning should be present on line 8, not line 1.

```ts
import "a";
import "b";


const a = "a";
const b = "b";

export { b, a };
```

Output:
```
/main.ts:1:1 [assist/source/organizeImports](https://biomejs.dev/assist/actions/organize-imports)  FIXABLE  ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━

  ℹ The imports and exports are not sorted.
  
  > 1 │ import "a";
      │ ^^^^^^^^^^^
    2 │ import "b";
    3 │ 
  
  ℹ Safe fix: Organize Imports (Biome)
  
    6 6 │   const b = "b";
    7 7 │   
    8   │ - export·{·b,·a·};
      8 │ + export·{·a,·b·};
    9 9 │   
```

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Kakadus** [comment]
Introduced in #9353

**Comment by ematipico** [comment]
@Conaclos 

I believe we need to do a better job at calculating the correct `text_range`

**Comment by Conaclos** [comment]
> I believe we need to do a better job at calculating the correct text_range

Could this conflict with biome-ignore comments? Although this could be ok since we now have ignore-comments for the entire file?

**Comment by ematipico** [comment]
@Conaclos  I don't know to be honest, at the moment they share the same range (the implementation is weird, by default `text_range` uses the range emitted by the diagnostic function).

https://github.com/biomejs/biome/blob/794f79c3d61d0010549f8f34b05c0aaa5eb2b1c3/crates/biome_js_analyze/src/assist/source/organize_imports.rs#L708-L729

So there's conflict right now. The bug is plain and simple: we emit the incorrect diagnostic range.
